sycl/source/detail/graph_impl.cpp

```diff
@@ -1155,9 +1156,48 @@ void exec_graph_impl::duplicateNodes() {
   MNodeStorage.insert(MNodeStorage.begin(), NewNodes.begin(), NewNodes.end());
 }
 
+void exec_graph_impl::update(std::shared_ptr<graph_impl> GraphImpl) {
+
+  if (MDevice != GraphImpl->getDevice()) {
+    throw sycl::exception(
+        sycl::make_error_code(errc::invalid),
+        "The graphs must have been created with matching devices.");
```

Comment:
Nit but slight preference for error wording that more clearly indicates the operation that failed (in this case updating), eg:

> Cannot update using a graph created with a different device

Reply:
Done :+1: